src/database/entitySpec.js

```diff
@@ -0,0 +1,59 @@
+const databaseStructure = {
+/** Users */
+  Users: {
+    id: Integer,
+    firstName: String,
+    lastName: String,
+    email: String,
+    password: String,
+    gender: String,
+    role: String, // portfolio in the company
+    isVerified: Boolean, // user signup verification, Default value is false
+    phoneNumber: String,
+    prefferedCurrency: String,
+
+  },
+
+
+  /** Requests */
+  Requests: {
+    id: Integer,
+    requesterId: Integer, // user id
+    managerId: Integer, // used to notify managers who also a user
+    reasons: String, // reasons for traveling
+    status: String, // valid values: open, accepted, rejected, Default value is open
+    tripId: Integer
+
+  },
+  /** Trips */
+  Trips: {
+    id: Integer,
+    tripType: String, // One-way or return trip
+    departureLocation: String,
+    destinationIds: Array(Integer), // an array of destinations
+    accommodationIds: Array(Integer), // an array of chosen accomodation linked to a destination
+    departureDate: Date,
+    returnDate: Date,
+
+  },
+
+  /** Destinations */
+  Destination: {
+    id: Integer,
+    name: String, // office locations or destinations, e.g Lagos, Nairobi, New York
+    country: String, // e.g. Nigeria
+    numberOfTimesVisited: Integer // number of times that the location has been visited
+  },
+
+  /** Accommodations */
+  Accommodations: {
+    id: Integer,
+    Locationid: Integer, // destination id
```

Comment:
Can't an accommodation be listed before a trip vis a vis destination is booked/known?

Reply:
every accommodation is tied to a destination. A destination has one or more accommodation so it should be defined before it